### PR TITLE
[HOTFIX] Different timestamp files for debug and release mode.

### LINKED ISSUE #2009: use two different timeouts in argparser ?
for developer and user? Otherwise the developers might never see the developer info, because she is also user (release binaries) and the other runs always invalidate the timer.

### PR DESCRIPTION
**Pull request by smehringer**
closes #2009

### CODE REVIEW

File: include/seqan/arg_parse/arg_parse_version_check.h [hunk 1]
```diff
@@ -108,7 +108,8 @@ struct VersionCheck
     std::string _version = "0.0.0";
     std::string _program;
     std::string _command;
-    std::string _path = _getPath();
+    const std::string _path = _getPath();
```
Comment (rrahn) [inline include/seqan/arg_parse/arg_parse_version_check.h]: Is this change necessary for the fix?

File: include/seqan/arg_parse/arg_parse_version_check.h [hunk 2]
```diff
@@ -122,6 +123,11 @@ struct VersionCheck
         errorStream(errorStream)
     {
         std::smatch versionMatch;
+#if defined(NDEBUG) || defined(SEQAN_TEST_VERSION_CHECK_)
+        _timestamp_filename = _path + "/" + _name + "_release.timestamp";
+#else
+        _timestamp_filename = _path + "/" + _name + "_debug.timestamp";
```
Comment (rrahn) [inline include/seqan/arg_parse/arg_parse_version_check.h]: maybe call it `_dev.timestamp` and `_usr.timestamp` to indicate that the one is meant for developers and the other for users.

File: include/seqan/arg_parse/arg_parse_version_check.h [hunk 3]
```diff
@@ -318,11 +324,11 @@ inline std::string _getPath()
 // Function _getFileTimeDiff()
 // ----------------------------------------------------------------------------
 
-inline double _getFileTimeDiff(std::string const & timestamp_filename)
+inline double _getFileTimeDiff(VersionCheck & me)
```
Comment (rrahn) [inline include/seqan/arg_parse/arg_parse_version_check.h]: please pass as `const&` if the state is not changed.

Comment (h-2) [inline include/seqan/arg_parse/arg_parse_version_check.h]: @rrahn means `const &` with a space in between

File: include/seqan/arg_parse/arg_parse_version_check.h (hunk 1 above)
Comment (smehringer) [inline include/seqan/arg_parse/arg_parse_version_check.h]: no sorry. I wondered why I did not declare this const and forgot to change it back.

File: tests/arg_parse/test_arg_parse_version_check.h [hunk 4]
```diff
@@ -63,7 +63,7 @@ const std::string TestVersionCheck_::APP_VERSION_FILENAME     = TestVersionCheck
                                                                 static_cast<std::string>(".version");
 const std::string TestVersionCheck_::APP_TIMESTAMP_FILENAME   = TestVersionCheck_::PATH + "/" +
                                                                 TestVersionCheck_::APP_NAME +
-                                                                static_cast<std::string>("_release.timestamp");
+                                                                static_cast<std::string>("_usr.timestamp");
```
Comment (rrahn) [inline tests/arg_parse/test_arg_parse_version_check.h]: Is the dev.timestamp tested as well?

Comment (smehringer) [inline tests/arg_parse/test_arg_parse_version_check.h]: No since it's only a name difference and no change in functionality?
I could choose either one and I chose to use the usr.timestamp filename.

**Comment by smehringer** [comment]
The searchjoin test failure is unrelated to my PR.
